pkg/kubelet/kubelet_pods.go

```diff
@@ -1234,12 +1234,6 @@ func (kl *Kubelet) GetKubeletContainerLogs(ctx context.Context, podFullName, con
 		return err
 	}
 
-	if kl.dockerLegacyService != nil {
-		// dockerLegacyService should only be non-nil when we actually need it, so
-		// inject it into the runtimeService.
-		// TODO(random-liu): Remove this hack after deprecating unsupported log driver.
```

Comment:
was it actually deprecated?

Reply:
Again, going back in my memory a bit here :)

In the commit, I wrote:
```
I'm not actually sure if this is safe... if it isn't, we
should be able to figure out an easy way to make it work with a similar
strategy to the one we are taking for dockershim.
```

Looking at this now... we use the `dockerLegacyService` only if `IsCRISupportedLogDriver` returns false. We return false if the log driver in use isn't one of the ones listed [here](https://sourcegraph.com/github.com/kubernetes/kubernetes@master/-/blob/pkg/kubelet/dockershim/docker_legacy_service.go#L122). Ultimately, that list contains `json-file`.

`json-file` is the default log driver for Docker as of 19.03 *https://github.com/moby/moby/blob/19.03/cmd/dockerd/config.go#L68), which is the latest validated docker version in master (https://sourcegraph.com/github.com/kubernetes/kubernetes@master/-/blob/vendor/k8s.io/system-validators/validators/docker_validator.go#L52:2).

So I _think_ it should be safe to delete this? 

cc @Random-Liu 

If we do decide that its safe to delete this, I can continue the clean up in a separate set of diffs (i.e. I think there may be some additional `dockershim` methods we can delete).

As I mention in the commit message, we can also keep this if we want - we can define a "nodocker" alternative which throws an error if you actually try to use it.

Reply:
put another way: if we have not officially deprecated support for some functionality, I don't think we can / should be just deleting it.

the comment here suggests that we were going to deprecate this, but I'd love to see evidence that we in fact did deprecate it already.

Reply:
@mattjmcnaughton looks like instead of deleting we need to add a method in some utility go file and have a dummy empty method when `dockerless` tag is active.

Reply:
Sounds like a plan :)